### PR TITLE
Feature/us 35/domain layer

### PR DESCRIPTION
**Pull request by CatCoder36**
**Description**
The model class was created to represent the users in the program.
Changes made:
- Creation of model directory in the usersmanagement package.
- Modified gitignore to ignore .idea files.

**CodeReview:** 

- @victor0villca 
- @daniel-espinoza-79

### CODE REVIEW

**Review by victor0villca — APPROVED** [review]
Approve, the domain layer is great.

File: backend/src/main/java/com/fake_orgasm/users_management/models/User.java [hunk 1]
```diff
@@ -36,9 +43,56 @@ public class User implements Comparable<User> {
...
+     * This method use the category as parameter of compare, return a positive
+     * value if the current user  has a higher category than the
+     * compared user, and return a negative value in the other case,
+     * return 0 when the user have the same category.
+     *
+     * @param o the object to be compared.
+     * @return int result of the comparative.
      */
     @Override
     public int compareTo(User o) {
```
Comment (daniel-espinoza-79) [inline backend/src/main/java/com/fake_orgasm/users_management/models/User.java:89]: The comparison should be by the name and if  both have the same name, it should be by the id

Comment (CatCoder36) [reply backend/src/main/java/com/fake_orgasm/users_management/models/User.java:99]: Solved

File: backend/src/main/java/com/fake_orgasm/users_management/models/User.java [hunk 2]
```diff
@@ -36,9 +43,77 @@ public class User implements Comparable<User> {
...
-        return 0;
+        int resultCompare = 0;
+        char[] nameUser = getFullName().toCharArray();
+        char[] nameUserCompared = o.getFullName().toCharArray();
```
Comment (daniel-espinoza-79) [inline backend/src/main/java/com/fake_orgasm/users_management/models/User.java:102]: You can use  the compareTo by default of a String for the fullName, 
getFullName().compareTo();

**Review by daniel-espinoza-79 — APPROVED** [review]

File: backend/src/main/java/com/fake_orgasm/users_management/models/User.java (hunk 2 above)
Comment (CatCoder36) [reply backend/src/main/java/com/fake_orgasm/users_management/models/User.java]: Fixed

**Pull request merged**